src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,384 @@
+
+        'connected_account_id': String
+            (Optional) The ID of the customer connected account to work with.
+
+        'customer_subnet_ids': Array of String
+            (Optional) List of subnet Ids
+
+    deployment_type: String
+        (Optional) Denotes if request is for a SingleAZ or a MultiAZ SDDC. Default is SingleAZ.
+        Possible values are: SingleAZ, MultiAZ
+
+    host_instance_type: String
+        (Optional) The instance type for the ESX hosts in the primary cluster of the SDDC.
+        Possible values are: i3.metal, r5.metal, i3en.metal
+
+    msft_license_config :
+        (Optional) Indicates the desired licensing support, if any, of Microsoft software.
+        It can be specified in the below format
+
+        .. code::
+
+            msft_license_config
+                - academic_license: False
+                - mssql_licensing: "DISABLED"
+                - windows_licensing: "DISABLED"
+
+             where,
+             academic_license - Flag to identify if it is Academic Standard or Commercial Standard License.
+             mssql_licensing - The status MSSQL licensing for this SDDC’s clusters.
+                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
+             windows_licensing - The status of Windows licensing for this SDDC’s clusters. Can be enabled, disabled, or customer’s.
+                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
+
+        Please refer the `VMC Doc about msft_license_config <https://developer.vmware.com/apis/vmc/v1.1/data-structures/MsftLicensingConfig/>`_
+
+    sddc_id: String As UUID
+        (Optional)If provided, will be assigned as SDDC id of the provisioned SDDC.
+
+    sddc_template_id : String As UUID
+        (Optional) If provided, configuration from the template will applied to the provisioned SDDC.
+
+    sddc_type: String
+        (Optional)Denotes the SDDC type, if the value is null or empty, the type is considered as default.
+
+    size: String
+        (Optional) The size of the vCenter and NSX appliances. “large” sddcSize corresponds to a ‘large’ vCenter
+        appliance and ‘large’ NSX appliance. ‘medium’ sddcSize corresponds to ‘medium’ vCenter appliance and
+        ‘medium’ NSX appliance. Value defaults to ‘medium’.
+        Possible values are: nsx_small, medium, large, nsx_large
+
+    skip_creating_vxlan : Boolean
+        (Optional) skip creating vxlan for compute gateway for SDDC provisioning
+
+    sso_domain : String
+        (Optional) The SSO domain name to use for vSphere users. If not specified, vmc.local will be used.
+
+    storage_capacity:  Integer As Int64
+        (Optional) The storage capacity value to be requested for the SDDC primary cluster, in GiBs. If provided,
+        instead of using the direct-attached storage, a capacity value amount of seperable storage will be used.
```

Comment:
Spelled `separable`, however, unless "separable storage" is an SDDC term I'm unfamiliar with, I think this should be rephrased for clarity.

Reply:
This definition of parameter is mentioned on API reference documentation, cross checked again to be sure.

Reply:
That's a typo in the upstream doc: 
![image](https://user-images.githubusercontent.com/189750/198088673-4b22949e-310f-4874-873b-570a74880533.png)

This should be fixed (not sure where else to notify - mentioned it on the feedback form of the docs page)

Reply:
Updated